src/CodeGen_LLVM.cpp

```diff
@@ -1614,6 +1614,12 @@ void CodeGen_LLVM::visit(const Load *op) {
 
     bool possibly_misaligned = (might_be_misaligned.find(op->name) != might_be_misaligned.end());
 
+    // If it's a Handle, load it as a uint64_t and then cast
+    if (op->type.is_handle()) {
```

Comment:
Hrm, good point r.e. loading all 64-bits not being necessary if we immediately truncate. This won't trigger on retrieving fields from a buffer_t, they get unpacked separately and don't use a Load node.